src/App.tsx

```diff
@@ -15,36 +23,99 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortBy {
+  'default',
+  'alpabet',
+  'length',
+}
+
+function getGoods(goodsFromServList: string[]): GoodsList {
+  return goodsFromServList.map((str, index) => ({ id: index, val: str }));
+}
+
+function prepGoods(goods: GoodsList, oper: SortBy, revOp: boolean): GoodsList {
+  let tempArr = getGoods(goodsFromServer);
+
+  if (oper === 0) {
+    return revOp ? tempArr.reverse() : tempArr;
+  }
+
+  tempArr = goods.sort((g1: GoodsFromServer, g2: GoodsFromServer) => {
```

Comment:
You should avoid direct sorting of the `goods` parameter as it can lead to unintended side effects. It's better to create a new array or copy the existing one before sorting.